### DISCUSSION TITLE
How to set `VCPKG_OSX_DEPLOYMENT_TARGET` when using vcpkg with cmake?

### DISCUSSION THREAD

**Discussion by JAicewizard** [microsoft/vcpkg#39635] [Q&A]
Hello, I have a project setup with CMAKE and I am using vcpkg to install the dependencies. On macos, the cmake project is build against macos version 11, but the dependencies build with vcpkg build against macos 14 (the host OS version) instead. This does not directly lead to errors, but it does bring up some warnings. Besides, I would like to target older versions as well.

Is there any way to specify the macos target version when invoking cmake?

**Comment by autoantwort** [comment]
This is a triplet variable (https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_osx_deployment_target), so create a triplet and set the variable in there and then use the triplet.

**Reply by unknown-address** [reply]
Hi there,

Slightly related to this question. Is there a way to set a variable that is both used in triplet files and CMakeLists.txt files? I would like to set a variable, e.g. MAC_OS_TARGET_VERSION, once and then be able to use it in triplet files and CMakeLists.txt files in my project.

Like so, for example:
in arm64-osx.cmake
```cmake
set(VCPKG_OSX_DEPLOYMENT_TARGET "${MAC_OS_TARGET_VERSION}")
```

in root CMakeLists.txt
```
set(CMAKE_OSX_DEPLOYMENT_TARGET ${MAC_OS_TARGET_VERSION})
```

As far as I understand triplet files are not evaluated as part of my project's CMake configure step, hence I can't set it in the root CMakeLists file. Is there a way to share variables between my project's scope and and vcpkg?

**Reply by autoantwort** [reply]
I guess you could set it as a env var, but that would break the computation of the abi of the triplet since only the file content gets hashed and that would be the same nevertheless how the env var is set
